config/c-code/tests.yml.j2

```diff
@@ -112,52 +117,128 @@ jobs:
         run: |
           pip install -U pip
           pip install -U setuptools wheel twine cffi
-
-{% if with_future_python %}
-      - name: Build %(package_name)s (%(future_python_version)s)
-        if: ${{ startsWith(matrix.python-version, '%(future_python_version)s') }}
-        run: |
-          # Next, build the wheel *in place*. This helps ccache, and also lets us cache the configure
-          # output (pip install uses a random temporary directory, making this difficult).
-          python setup.py build_ext -i
-          python setup.py bdist_wheel
-          # Also install it, so that we get dependencies in the (pip) cache.
-          pip install -U 'faulthandler; python_version == "2.7" and platform_python_implementation == "CPython"'
-          pip install --pre .[test]
+{% if gha_additional_build_dependencies %}
+  {% for line in gha_additional_build_dependencies %}
+          pip install -U %(line)s
+  {% endfor %}
 {% endif %}
-{% for kind in ('Python 3.10 and 3.11 on MacOS', 'all other versions') %}
+
+{% for kind in ('macOS x86_64, Python 3.8+',
+                'macOS arm64, Python 3.8+',
+                'macOS universal2, Python 3.8+',
+                'all other versions') %}
       - name: Build %(package_name)s (%(kind)s)
         if: >
-  {% if kind == 'Python 3.10 and 3.11 on MacOS' %}
+  {% if kind == 'macOS x86_64, Python 3.8+' %}
           startsWith(runner.os, 'Mac')
-          && (startsWith(matrix.python-version, '3.10')
-              || startsWith(matrix.python-version, '3.11'))
+          && !(startsWith(matrix.python-version, 'pypy')
+               || matrix.python-version == '2.7'
+               || matrix.python-version == '3.5'
```

Comment:
This looks okay as this list will shrink when we stop supporting Python < 3.6. (Maybe we could even think about dropping Python 3.6 support.)

Reply:
I wanted to make sure it's a list that doesn't need growing as new Python versions are releases. I'm still unhappy that there's apparently no way to safely express something like `matrix.python-version < '3.8'`, something that understands versioning.